### PR TITLE
Bookstore: Use Redux in React components

### PR DESCRIPTION
**Pull request by mirouhml**
- Keep state immutable in reducers!
- Import Redux Provider and in the store of the main component of the React app.
- Connect the app to the Redux store.
- Make sure that list of books displays books from Redux store.
  - Use `useSelector` to consume the state.
- In the React component `Books.js`, add funcionality for "Add new" button click:
  - Import Redux `useDispatch`.
  - Set the data inputs in the local React state.
  - Once the new book object is ready to be submitted to Redux store, dispatch a corresponding action.
- In the React component `Books.js`, add a funcionality for "Remove" button click:
  - Import Redux useDispatch.
  - Dispatch a corresponding action.
- In the React component `Categories.js`, add a funcionality for "Check status" button click:
  - Import Redux useDispatch.
  - Dispatch a corresponding action.

### CODE REVIEW

File: src/components/Form.js [hunk 1]
```diff
@@ -1,21 +1,48 @@
...
-  </div>
-);
+const Form = (props) => {
+  const { addBook } = props;
+  return (
+    <div>
+      <h2>Add a new book</h2>
+      <form>
+        <input type="text" placeholder="Book title" id="book-title" />
+        <input type="text" placeholder="Author" id="book-author" />
```
Comment (Meltrust) [inline src/components/Form.js]: According to the project requirements:

image

Therefore:

- [x] Please, make these inputs use a React local state in this component.  They should have their inputs that update the local state every time they update (onChange).



## Explanation

We need to make the inputs update the local state (not Redux) every time they update (every keypress from the user).  This is a common React practice with forms.

## Hints

You can import `useState` and use something like `setTitle` and `setAuthor` for the new `onChange` properties on each `<input>`.

Like this:

### Set the state:

```JAVASCRIPT
import React, { useState } from 'react';

const [title, setTitle] = useState('');
const [author, setAuthor] = useState('');
```

### On your inputs:

```JAVASCRIPT
value={title}
onChange={(event) => setTitle(event.target.value)}

value={author}
onChange={(event) => setAuthor(event.target.value)}
```

_Note: This is just an example intended to help.  Your own implementation may vary._

That way, we will meet this particular requirement, and your inputs will update your local state on each keystroke! 💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @mirouhml,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To highlight:

- Nice use of Redux, local state, and props✔️ 

You can do it

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Form.js (hunk 1 above)
Comment (mirouhml) [reply src/components/Form.js]: @Meltrust thanks for the detailed explanation, I just added it!

**Review by Meltrust — APPROVED** [review]
Hi @mirouhml,


## Wow, you did it 🎉
Brilliant
## Thank you for implementing the local state for your inputs, as requested in the previous review 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mirouhml** [comment]
@Meltrust thanks so much for the help!

**Pull request merged**